File: 442-find-all-duplicates-in-an-array/find-all-duplicates-in-an-array.py
class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        mp = {}

        for i in range(len(nums)):
            if nums[i] in mp:
                mp[nums[i]]+=1
            else:
                mp[nums[i]]=1

        l1 = []

        for i in range(len(nums)):
            if mp[nums[i]]>1:
                l1.append(nums[i])
                mp[nums[i]]=0
        
        return l1

File: 442-find-all-duplicates-in-an-array/find-all-duplicates-in-an-array.cpp
class Solution {
public:
    vector<int> findDuplicates(vector<int>& nums) {
        
        map<int,int>mp;

        for(int i=0;i<nums.size();i++){
            mp[nums[i]]++;
        }
        vector<int>v1;
        for(int i=0;i<nums.size();i++){
            if(mp[nums[i]]>1){
                v1.push_back(nums[i]);
                mp[nums[i]]=0;
            }
        }

        return v1;

    }
};